shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Tx.hs

```diff
@@ -361,3 +365,33 @@ extractKeyHashWitnessSet credentials = foldr accum Set.empty credentials
   where
     accum (KeyHashObj hk) ans = Set.insert (asWitness hk) ans
     accum _other ans = ans
+
+
+validateScriptB ::
+  Era era =>
+  Script era ->
+  Set (KeyHash 'Witness era) ->
+  TxBody era ->
+  Bool
+validateScriptB (MultiSigScript msig) vhks _tx = evalNativeMultiSigScript msig vhks
+validateScriptB(TimelockScript timel) vhks tx = evalFPS timel vhks tx
```

Comment:
I would suspect evalFPS needs a better name even though that's what I used. 
Looks good otherwise (+ Nick's comments)

Reply:
as long as we keep the names the same between the spec and the implementation I'm happy :)